src/viam/sdk/services/mlmodel/private/proto.cpp

```diff
@@ -338,6 +341,145 @@ class tensor_to_pb_value_visitor : public boost::static_visitor<::grpc::Status>
+        target_->mutable_int16_tensor()->mutable_data()->Clear();
+        target_->mutable_int16_tensor()->mutable_data()->Resize(num32s, 0);
+        std::memcpy(target_->mutable_int16_tensor()->mutable_data()->mutable_data(),
+                    t.begin(),
+                    t.size() * sizeof(std::int16_t));
+    }
+
+    void dispatch_(const MLModelService::tensor_view<std::uint16_t>& t) const {
+        // TODO: LE specific. See above comment.
+        const int num32s =
+            static_cast<int>((t.size() + 1) * sizeof(std::uint16_t) / sizeof(std::uint32_t));
+        target_->mutable_uint16_tensor()->mutable_data()->Clear();
+        target_->mutable_uint16_tensor()->mutable_data()->Resize(num32s, 0);
+        std::memcpy(target_->mutable_uint16_tensor()->mutable_data()->mutable_data(),
+                    t.begin(),
+                    t.size() * sizeof(std::uint16_t));
+    }
+
+    void dispatch_(const MLModelService::tensor_view<std::int32_t>& t) const {
+        target_->mutable_int32_tensor()->mutable_data()->Clear();
+        target_->mutable_int32_tensor()->mutable_data()->Add(t.begin(), t.end());
+    }
+
+    void dispatch_(const MLModelService::tensor_view<std::uint32_t>& t) const {
+        target_->mutable_uint32_tensor()->mutable_data()->Clear();
+        target_->mutable_uint32_tensor()->mutable_data()->Add(t.begin(), t.end());
+    }
+
+    void dispatch_(const MLModelService::tensor_view<std::int64_t>& t) const {
+        target_->mutable_int64_tensor()->mutable_data()->Clear();
+        target_->mutable_int64_tensor()->mutable_data()->Add(t.begin(), t.end());
+    }
+
+    void dispatch_(const MLModelService::tensor_view<std::uint64_t>& t) const {
+        target_->mutable_uint64_tensor()->mutable_data()->Clear();
+        target_->mutable_uint64_tensor()->mutable_data()->Add(t.begin(), t.end());
+    }
+
+    void dispatch_(const MLModelService::tensor_view<float>& t) const {
+        target_->mutable_float_tensor()->mutable_data()->Clear();
+        target_->mutable_float_tensor()->mutable_data()->Add(t.begin(), t.end());
+    }
+
+    void dispatch_(const MLModelService::tensor_view<double>& t) const {
+        target_->mutable_double_tensor()->mutable_data()->Clear();
+        target_->mutable_double_tensor()->mutable_data()->Add(t.begin(), t.end());
+    }
+
+    ::viam::service::mlmodel::v1::FlatTensor* target_;
+};
+
+template <typename T>
+MLModelService::tensor_views make_sdk_tensor_from_api_tensor_t(const T* data,
+                                                               std::size_t size,
+                                                               std::vector<std::size_t>&& shape,
+                                                               tensor_storage* ts) {
+    if (!data || (size == 0) || shape.empty()) {
+        std::ostringstream message;
+        message << "Empty or zero length data or shape";
+        throw std::invalid_argument(message.str());
```

Comment:
(q) just for my own clarity, what is `message` doing here? Naively it looks like we're adding a string and then immediately unpacking that string.

Reply:
@stuqdog - Merely copy paste from other places where I was adding some state to the string, but here I'm not. I'll remove it.